### PR TITLE
Per repo token based on installation_id

### PR DESCRIPTION
**Pull request by piotr-travisci**

### CODE REVIEW

File: lib/travis/github_apps.rb [hunk 1]
```diff
@@ -43,6 +44,7 @@ def initialize(installation_id, config = {})
 
       @accept_header       = config.fetch(:accept_header, "application/vnd.github.machine-man-preview+json")
       @installation_id     = installation_id
+      @repository_id       = repository_id
```
Comment (BanzaiMan) [inline lib/travis/github_apps.rb]: It is not clear to me by reading the code what kind of data `repository_id` is supposed to be passed when initializing this object.

File: lib/travis/github_apps.rb [hunk 2]
```diff
@@ -131,6 +134,10 @@ def fetch_new_access_token
       github_access_token
     end
 
+    def repositories_list
+      @repositories_list ||= repository_id.kind_of?(Array) ? repository_id.compact.reject(&:empty?).map(&:to_s) : [repository_id.to_s]
```
Comment (BanzaiMan) [inline lib/travis/github_apps.rb]: Is it supposed to be an array? Or a single String, or something else? If it's an array, it shouldn't be named `repository_id`, which implies that it is a scalar value.

**Review by BanzaiMan — CHANGES_REQUESTED** [review]
Needs some clarifications in the code's intent. Also needs specs.

**Comment by BanzaiMan** [comment]
Also, be sure to squash the commits when merging.

File: lib/travis/github_apps.rb (hunk 2 above)
Comment (piotr-travisci) [reply lib/travis/github_apps.rb]: It was possible to pass number or array, but you are right. We should keep it simple, so I refactored it.

**Review by BanzaiMan — APPROVED** [review]

**Pull request merged**